### PR TITLE
Docker fixes

### PR DESCRIPTION
**Pull request by slustig210**
- runs npm ci instead of npm install when building image so that it matches our package-lock.json
- runs migrate before creating container and then only generate in the container 😄 actually very cool 
- updated prisma
- created a persistent named node_modules volume
- created a migrate_reset script to drop all tables and create new ones

### CODE REVIEW

File: migrate_reset.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+docker-compose up -d db
+export DATABASE_URL='postgresql://postgres:docker@urefer-db:5432/urefer?schema=public'
+cd ./backend
+npx prisma migrate reset --skip-seed # seeding will be done by run...?
```
Comment (slustig210) [inline migrate_reset.sh:4]: thoughts on having this? and on skipping seed?

File: migrate_reset.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+docker-compose up -d db
+export DATABASE_URL='postgresql://postgres:docker@localhost:5432/urefer?schema=public'
+cd ./backend
+npx prisma migrate reset --skip-seed # seeding will be done by run...?
```
Comment (slustig210) [inline migrate_reset.sh:4]: thoughts on have the migrate_reset file and if it should seed or not?

Comment (slustig210) [reply migrate_reset.sh:4]: if u didnt know migrate reset drops all the tables and creates empty tables and then applies all migrations

**Review by Danny123234 — APPROVED** [review]
Holy wow

File: migrate_reset.sh (hunk 2 above)
Comment (rdonati) [reply migrate_reset.sh:4]: @slustig210 I think seeding is reasonable 👍

**Pull request merged**